acceptancetests/jujupy/client.py

```diff
@@ -2262,7 +2353,24 @@ def _get_full_path(juju_path):
         return os.path.abspath(juju_path)
 
 
-def client_from_config(config, juju_path, debug=False, soft_deadline=None):
+def decide_client(client_type):
```

Comment:
Moving ```CaasClient``` within ```ModelClient``` removes a lot of this complexity.

Reply:
I think we have already had different clients, I suggest we should refactor `ClientType.Fake` related code as well.